contracts/arbos/builtin/ArbSys.sol

```diff
@@ -22,6 +22,9 @@ interface ArbSys {
     // able to call it
     function getStorageAt(address account, uint256 index) external view returns (uint256);
 
+    // Return true if the caller of this was called directly from L1
+    function calledFromL1() external view returns (bool);
```

Comment:
I think this would be better framed as `isTopLevel` or `isTopLevelCall`. Other than this this looks good to me

Reply:
Renamed to `isTopLevelCall`.